students/ericstreit/lessons/lesson09/assignment/logging_assignment/charges_calc.py

```diff
@@ -0,0 +1,171 @@
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+
+def logger_toggle(func):
+    """ a timer decorator """
+    if (ARGS.debug) == 0:
+        # DO NOTHING
+        # Special thanks to:
+        # https://stackoverflow.com/questions/14636350/toggling-decorators
+        return func
+
+    else:
+        def set_level(*args, **kwargs):
+            """ nested function for timing functions """
+
+            # start = time.time()
+            value = func(*args, **kwargs)
+            # end = time.time()
+            # runtime = end - start
+            # print(f' The runtime for {func.__name__} took {runtime} seconds to complete')
+            return value
+    return set_level
```

Comment:
Decorators are run at import time, so the decorator is not going to know anything about ARGS. That is the tricky bit of this one. So, the decorator has to manipulate logging itself, and that will affect only the decorated functions, as long as you change logging before and return it back after you call the function in the decorator.